src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -189,22 +194,22 @@ public static void SpinWait(int iterations)
         /// </summary>
         public bool IsBackground
         {
-            get => IsBackgroundNative();
+            get => GetIsBackground(GetNativeHandle()) != Interop.BOOL.FALSE;
             set
             {
-                SetBackgroundNative(value);
+                SetIsBackground(GetNativeHandle(), value ? Interop.BOOL.TRUE : Interop.BOOL.FALSE);
                 if (!value)
                 {
                     _mayNeedResetForThreadPool = true;
                 }
             }
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private extern bool IsBackgroundNative();
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "ThreadNative_GetIsBackground")]
```

Comment:
This was FCall without HMF before. It is QCall with full GC transition now. The API is going to be a lot slower now. This is an API that can be called frequently. I do not think we want to be regressing its perf.